### PR TITLE
fix: dynamics examples, benchmarks, docs, and deprecation warnings

### PR DESCRIPTION
**Pull request by zubatyuk**
## Description

Fix pre-existing bugs across dynamics examples, benchmarks, and documentation that prevented examples and benchmarks from running, along with deprecation warning cleanup and doc cross-reference corrections.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

N/A

### Examples (`examples/dynamics/`)
- Remove invalid `device=` kwarg from `fire_step()`/`fire2_step()` calls in 6 example files (these functions do not accept a `device` parameter)
- Fix broken import in `07_fire_variable_cell.py` (`_langevin_utils` -> `_dynamics_utils`)
- Pre-allocate KE/temperature scratch buffers in `MDSystem` and `BatchedMDSystem` to eliminate warp allocations in hot loops
- Fix `compute_temperature` call signature: `num_atoms=` -> `num_atoms_per_system=` to match actual API

### Benchmarks (`benchmarks/dynamics/`)
- Fix `benchmark_fire_compare.py`: replace non-existent `NvalchemiopsLJModel`/`create_lj_system` imports with `create_fcc_argon`; use `torch.stack` instead of `torch.cat` for batched cells (correct `(B, 3, 3)` shape)
- Fix `shared_utils.py` `NvalchemiOpsBenchmark`:
  - Add `__getattr__` delegation to `self.system` (replaces broken `self.model` references throughout variable-cell FIRE/FIRE2 methods)
  - Add `_compute_forces` bridge method (was called but never defined)
  - Forward LJ parameters (`epsilon`/`sigma`/`cutoff`/`skin`/`switch_width`/`half_neighbor_list`) to unbatched `MDSystem` (was already done for batched)
  - Fix `self.device` (torch.device) -> `self.wp_device` (string) for all warp API calls
  - Fix stale position/cell sync in `MDSystem._rebuild_neighbors` and `BatchedMDSystem._rebuild_neighbors`
  - Remove dead interval-based neighbor rebuild code (variables were never initialized, displacement-based check already handles rebuilds)
  - Add `self.wp_batch_idx = None` to `MDSystem.__init__` for uniform attribute access

### Library (`nvalchemiops/`)
- Fix `cell_list` import in `nvalchemiops.neighborlist` via `__getattr__` with `DeprecationWarning` (was resolving to the warp-kernel submodule instead of the torch function)
- Replace deprecated `warp.types.Any` with `typing.Any` in electrostatics kernels (`dsf.py`, `pme_kernels.py`)

### Documentation
- Update migration guide with `cell_list` deprecation mapping
- Fix Sphinx cross-references pointing to non-existent `~nvalchemiops.torch.neighbors.unbatched.*` and `~nvalchemiops.torch.neighbors.batched.*` modules (corrected to `cell_list`/`naive`/`batch_cell_list`/`batch_naive`)
- Fix markdownlint line-length violations in `docs/userguide/about/intro.md` and `docs/benchmarks/neighborlist.md`

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [ ] New tests added for new functionality meets coverage expectations?
- [x] All pre-commit hooks pass
- [x] All dynamics examples run without errors (06-11)
- [x] All dynamics benchmarks run successfully (`benchmark_fire_compare`, `benchmark_opt_single`, `benchmark_opt_batch`)

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [x] I have updated the documentation (if applicable)

## Additional Notes

All changes fix pre-existing bugs or address API signature mismatches. No new features or behavioral changes are introduced. The `NvalchemiopsLJModel` class and `create_lj_system` function referenced in `benchmark_fire_compare.py` were never defined in `shared_utils.py`, meaning the benchmark was broken from the start. The variable-cell FIRE/FIRE2 benchmark methods referenced `self.model` which also never existed. These are now properly routed through `self.system` via `__getattr__` delegation.

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct pull request reviews. We encourage contributors to read and consider suggestions made by Greptile, but note that human maintainers will provide the necessary reviews for merging: Greptile's comments are **not** a qualitative judgement of your code, nor is it an indication that the PR will be accepted/rejected. We encourage the use of emoji reactions to Greptile comments, depending on their usefulness and accuracy.

### CODE REVIEW

File: examples/dynamics/_dynamics_utils.py [hunk 1]
```diff
@@ -1011,26 +1016,25 @@ def update_cell(self, cell: wp.array) -> None:
...
         compute_temperature(
             kinetic_energy=ke,
-            temperature=temp,
-            num_atoms=self.num_atoms,
-            device=self.device,
+            temperature=self._temp_buf,
+            num_atoms_per_system=self.wp_num_atoms_per_system,
         )
-        return temp
+        return self._temp_buf
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply examples/dynamics/_dynamics_utils.py:1037]: Callers consume the value immediately, so there's no actual bug.

File: benchmarks/dynamics/benchmark_fire_compare.py [hunk 2]
```diff
@@ -61,75 +60,45 @@
...
     """Create a NvalchemiOpsBenchmark for a perturbed LJ system."""
     torch.manual_seed(seed)
-    positions, cell, masses, velocities = create_lj_system(
-        num_atoms=num_atoms,
-        lattice_constant=5.26,
-        temperature=300.0,
-        device="cuda",
-        dtype=torch.float64,
-    )
+    num_cells = int((num_atoms // 4 + 1) ** (1.0 / 3.0))
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply benchmarks/dynamics/benchmark_fire_compare.py:63]: By design -- FCC only supports 4n^3 atoms. actual_atoms (line 67) already tracks the true count.

File: benchmarks/dynamics/shared_utils.py [hunk 3]
```diff
@@ -1864,15 +1866,57 @@ def __init__(
...
+                cutoff=cutoff,
+                skin=skin,
+                switch_width=switch_width,
+                half_neighbor_list=half_neighbor_list,
                 device=positions.device,
                 dtype=positions.dtype,
             )
 
+        self.neighbor_rebuild_interval = neighbor_rebuild_interval
+        self._steps_since_rebuild = 0
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply benchmarks/dynamics/shared_utils.py:1878]: Intentional - retained for API compat, displacement-based check handles rebuilds.

File: nvalchemiops/neighborlist/__init__.py [hunk 4]
```diff
@@ -138,15 +138,14 @@ def __getattr__(name: str):  # pragma: no cover
...
 
-            def neighbor_list(*args, **kwargs):
-                """Raise a `RuntimeError` if we can't use the new API with torch."""
+            def _missing(*args, **kwargs):
                 raise RuntimeError(
-                    "PyTorch is required to use the previous `neighbor_list` API."
+                    f"PyTorch is required to use the previous `{name}` API."
                     " Please install via `pip install 'nvalchemiops[torch]'`"
-                    " and import from `nvalchemiops.torch.neighbors.neighbor_list` instead."
+                    f" and import from `nvalchemiops.torch.neighbors.{name}` instead."
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply nvalchemiops/neighborlist/__init__.py]: Fixed in 90569460f28c237af545b0d000474b46ed3dafd0 -- message now reads `from nvalchemiops.torch.neighbors import cell_list`.

**Review by dallasfoster — APPROVED** [review]
